### PR TITLE
HHH-13673 Cryptic error when providing import.sql file without a terminal char at the end of each line

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-13673

### CODE REVIEW

File: hibernate-core/src/main/antlr/sql-stmt.g [hunk 1]
```diff
@@ -40,6 +40,10 @@ options {
 
     public void throwExceptionIfErrorOccurred() {
         if ( errorHandler.hasErrors() ) {
+        	String errorMessage = errorHandler.getErrorMessage();
+        	if(errorMessage.contains("expecting STMT_END")) {
+        		throw new StatementParserException( "Import script Sql statements do not terminate with a proper terminal char. Supported terminal chars are ';' , ' ' , '\\t ' , '\\r' , '\\n' "  );
```
[the remark this answers is not in the document]
Comment (Sanne) [reply hibernate-core/src/main/antlr/sql-stmt.g]: Also, do we really support just using `' '` ? a blank space? I'm surprised the parser can deal with that.

**Comment by sebersole** [comment]
Out of curiosity, have you tried recognizing EOF?  I get that the error message should be better, but should this really be an error in the first place?

File: hibernate-core/src/main/antlr/sql-stmt.g (hunk 1 above)
[the remark this answers is not in the document]
Comment (dreab8) [reply hibernate-core/src/main/antlr/sql-stmt.g]: in sql-stmt.g whe define
```
STMT_END
    : ';' ( '\t' | ' ' | '\r' | '\n' )*
    ;
```
not sure if we really support the blank space, I have to test it

**Comment by dreab8** [comment]
@sebersole  the exception happens only when using the `MultipleLinesSqlCommandExtractor` and in case of multiple statements not sure the EOF is correct, with the `SingleLineSqlCommandExtractor` we do not have an error during parsing.

File: hibernate-core/src/main/antlr/sql-stmt.g (hunk 1 above)
[the remark this answers is not in the document]
Comment (sebersole) [reply hibernate-core/src/main/antlr/sql-stmt.g]: TBH I'm surprised we support anything other than `;` or `EOF` as the statement terminator

[the remark this answers is not in the document]
Comment (Sanne) [reply hibernate-core/src/main/antlr/sql-stmt.g]: > is it `\t ` or `\t` ?

@dreab8 I meant that the error message has an extra space in `'\\t '`

> in sql-stmt.g whe define
> 
> ```
> STMT_END
>     : ';' ( '\t' | ' ' | '\r' | '\n' )*
>     ;
> ```
> 
> not sure if we really support the blank space, I have to test it

Doesn't that mean ';' followed by an unlimited number of whitespace characters?

[the remark this answers is not in the document]
Comment (dreab8) [reply hibernate-core/src/main/antlr/sql-stmt.g]: @Sanne you are right

[the remark this answers is not in the document]
Comment (dreab8) [reply hibernate-core/src/main/antlr/sql-stmt.g]: I'm going to fix the message

[the remark this answers is not in the document]
Comment (dreab8) [reply hibernate-core/src/main/antlr/sql-stmt.g]: what about "Import script Sql statements must terminate with a ';' char" ?

[the remark this answers is not in the document]
Comment (Sanne) [reply hibernate-core/src/main/antlr/sql-stmt.g]: :+1: sounds appropriate. Thanks!

File: hibernate-core/src/main/antlr/sql-stmt.g [hunk 2]
```diff
@@ -40,6 +40,10 @@ options {
 
     public void throwExceptionIfErrorOccurred() {
         if ( errorHandler.hasErrors() ) {
+        	String errorMessage = errorHandler.getErrorMessage();
+        	if(errorMessage.contains("expecting STMT_END")) {
```
Comment (Sanne) [inline hibernate-core/src/main/antlr/sql-stmt.g]: BTW your code formatter seems off? missing spaces here..

File: hibernate-core/src/main/antlr/sql-stmt.g [hunk 3]
```diff
@@ -40,6 +40,10 @@ options {
...
+        		throw new StatementParserException( "Import script Sql statements must terminate with a ';' char"  );
```
Comment (Sanne) [inline hibernate-core/src/main/antlr/sql-stmt.g]: and a space too many here, just before closing '  );'

Comment (Sanne) [reply hibernate-core/src/main/antlr/sql-stmt.g]: ah it's because it's a `.g` file, of course :) ok...

**Comment by Sanne** [comment]
merged. thanks!

**Pull request closed without merging**